### PR TITLE
Iframe script eval

### PR DESCRIPTION
**Pull request by peyerluk**

### CODE REVIEW

**Comment by peyerluk** [comment]
Fixed in upstream: https://github.com/jquery/jquery/commit/22449eb968622c2e14d6c8d8de2cf1e1ba4adccd

File: src/core.js [hunk 1]
```diff
@@ -250,12 +250,13 @@ jQuery.extend({
...
-	globalEval: function( code ) {
-		var script = document.createElement( "script" );
+	// Evaluates a script in a global context (or the passed context)
+	globalEval: function( code, context ) {
+		context = context || document;
+		var script = context.createElement( "script" );
 
 		script.text = code;
-		document.head.appendChild( script ).parentNode.removeChild( script );
+		context.head.appendChild( script ).parentNode.removeChild( script );
```
Comment (marcbachmann) [inline src/core.js]: I thought that bug would be more difficult to fix :)

Comment (marcbachmann) [inline src/core.js]: Ah, they applied your fix to the core. nice.

Comment (gabrielhase) [inline src/core.js]: Yes, but they are not making a release...

Comment (marcbachmann) [inline src/core.js]: It will be in jquery 3, right?

Comment (gabrielhase) [inline src/core.js]: Yes, of course. When it comes out, somewhen in 2023.

Comment (marcbachmann) [inline src/core.js]: :laughing:

**Comment by masone** [comment]
This seems to have been reverted 5 days ago https://github.com/jquery/jquery/commit/6680c1b29ea79bf33ac6bd31578755c7c514ed3e

**Comment by masone** [comment]
Moved to DOMEval https://github.com/jquery/jquery/blob/6680c1b29ea79bf33ac6bd31578755c7c514ed3e/src/core/DOMEval.js

**Comment by masone** [comment]
It looks to me as your changes have been applied. @peyerluk can you please check if this fix is obsolete? Or tell me what I can test against the new jquery version to make sure it works.

**Comment by peyerluk** [comment]
This is the pr where the fix landed: https://github.com/upfrontIO/livingdocs-editor/pull/229/files#diff-ad832d18dd36aadde8300a7ff3d7db10

Unfortunately there is not automated test. I just checked if our jquery was loaded: https://github.com/upfrontIO/livingdocs-editor/pull/229/files#diff-ad832d18dd36aadde8300a7ff3d7db10R5 (interesting that the discussion you referred to is about exposing the second param in `globalEval`)

To test it manually one has to add a script in a doc-html and check if the script is executed in the iframe. Maybe we can write a proper test.

Is jQuery 3 around the corner?

**Comment by masone** [comment]
Argh, you're right. The DOMEval thing is not on 2.2-stable, only on master

**Comment by masone** [comment]
Whoop! This has made it in fridays [2.2 release](http://blog.jquery.com/2016/01/08/jquery-2-2-and-1-12-released/)! 

Your [commit is tagged 2.2](https://github.com/jquery/jquery/commit/22449eb968622c2e14d6c8d8de2cf1e1ba4adccd) and [the issue](https://github.com/jquery/jquery/issues/1757#ref-issue-117150910) is referenced from the release issue.

**Comment by marcbachmann** [comment]
:clap:

**Comment by peyerluk** [comment]
:cry: It got reverted. Did not land in 2.2

Reason: API change (signature change)
Name: Manipulation: execute scripts from iframe in the iframe's context
Commit: 22449eb
Issue: #1757
PR: #2696
Note: depends on "Ajax: Always use script injection in globalEval"

(see: https://github.com/jquery/jquery/issues/2723)

**Comment by boennemann** [comment]
arghs :( Is there a strategy to patch this from outside of jQuery, so we don't need to fork it?

**Comment by masone** [comment]
Argh!!!

**Comment by peyerluk** [comment]
This is not used anymore by the framework.

**Pull request closed without merging**